File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-1.cpp
class Solution {
public:
    bool mergeTriplets(vector<vector<int>>& triplets, vector<int>& target) {
        bool x=false,y=false,z=false;
        for (const auto& t: triplets){
            x|=((target[0]==t[0]) && (t[1]<=target[1]) &&(t[2]<=target[2]));
            y|=((target[0]>=t[0]) && (t[1]==target[1]) &&(t[2]<=target[2]));
            z|=((target[0]>=t[0]) && (t[1]<=target[1]) &&(t[2]==target[2]));
            if (x &&y && z)
                return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        res,n= float('inf'),len(nums)
        l,r=0,0
        curr=0
        for r in range(n):
            curr+=nums[r]
            while curr >=target and l<=r: 
                res=min(res,r-l+1)
                curr -= nums[l]
                l+=1
            
        return res if res !=  float('inf') else 0